caddytls/client.go

```diff
@@ -60,6 +61,9 @@ var newACMEClient = func(config *Config, allowPrompts bool) (*ACMEClient, error)
 		return nil, fmt.Errorf("%s: insecure CA URL (HTTPS required)", caURL)
 	}
 
+	// Set timeout by replacing ACME's http.Client
+	acme.HTTPClient = http.Client{Timeout: HTTPTimeout}
```

Comment:
This creates a new http client every time we perform an ACME request. Actually, what if we just changed the timeout value? `acme.HTTPClient.Timeout = HTTPTimeout` - then we don't have to replace the whole client each time.